docs/react/migration/migration-cra.md

```diff
@@ -0,0 +1,243 @@
+
+- Create an Nx workspace for a React application
+- Migrate a React application into your Nx workspace
+- Convert a CRA scripts for use in Nx
+- Create a library and use it in your application
+
+For this example, you’ll be migrating the default CRA typescript template app into an Nx workspace. This is the code that is generated when you run `yarn create react-app webapp --template typescript`.
+
+There is also a [repo](https://github.com/jaysoo/migrate-cra-to-nx) that shows the finished result of this guide and for each step a [diff](https://github.com/isaacplmann/migrate-cra-to-nx/commits/stepbystep) will be provided to see the exact code changes that occur for that step.
+
+## 1. Create your workspace
+
+To start migrating your app, create an Nx workspace:
+
+```bash
+npx create-nx-workspace acme --appName=webapp --preset=react --style=css --nx-cloud
+```
+
+**Note:** Replace `acme` with your organization's npm scope. This will be used when importing workspace projects. You can also replace `webapp` with a different name -- you can have more than one app in an Nx workspace.
+
+[View the code changes](https://github.com/isaacplmann/migrate-cra-to-nx/commit/3f2a00924e1c7e42875493f9d71b067d7ca69fe1)
+
+## 2. Add npm packages to your workspace to support CRA
+
+We'll need to add a few dependencies to the Nx workspace that are needed to allow CRA to function.
+
+```bash
+yarn add --dev react-scripts @testing-library/jest-dom eslint-config-react-app react-app-rewired
+```
+
+**Note:** The `react-app-rewired` package allows us to customize the webpack config without ejecting.
+
+[View the code changes](https://github.com/isaacplmann/migrate-cra-to-nx/commit/31fa9a327a939abe0fa420130760bde974b2f6fe)
+
+## 3. Replace code generated by Nx with the CRA app
+
+The source code for each app in an Nx workspace should be contained within the folder of a generated app. The `create-nx-workspace` command from step 1 created an app folder at `apps/webapp` that we can use to contain the CRA app. Delete the existing contents and copy over the CRA app code.
+
+```bash
+rm -rf apps/webapp/* apps/webapp/{.babelrc,.browserslistrc}
+cp -r /path/to/cra-app/{README.md,package.json,tsconfig.json,src,public} apps/webapp
+```
+
+Replace `/path/to/cra-app` with the actual path to your CRA app on your machine.
+
+[View the code changes](https://github.com/isaacplmann/migrate-cra-to-nx/commit/a8ad934998949a760e5b62782689683cb469dd7c)
+
+## 4. Add CRA commands to workspace.json
+
+Find the `webapp` section inside `/workspace.json` and replace it with the following text. The `architect` section defines commands that can be executed by Nx and what Nx should do when those commands are triggered. The [run-commands](/{{framework}}/plugins/workspace/builders/run-commands) builder executes a terminal command. `cwd` specifies the directory the command should be executed from and `outputs` lets Nx know which files are modified when the command is executed.
```

Comment:
We should use the `run-commands` schematic here instead since it will be part of Nx 10 (already available in beta.3). No manual JSON edits necessary. :)